### PR TITLE
docs: Add Azure example configuration snippets

### LINKED ISSUE #8301: [Docs] Requesting examples for Azure Storage
E-mail request from a user

Hello team
Kindly check as there is no example docs for using Azure Storage
https://grafana.com/docs/loki/latest/configuration/examples/

**Describe the solution you'd like**
Create configuration example for Azure Storage.  If there is already an example configuration somewhere, add it to the topic.

### PR DESCRIPTION
**Pull request by paul1r**
**What this PR does / why we need it**:
Adds configuration snippets related to Azure to the examples.

**Which issue(s) this PR fixes**:
Fixes #8301 

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: docs/sources/configure/examples/configuration-examples.md [hunk 1]
```diff
@@ -402,3 +402,47 @@ memberlist:
...
+  azure:
+    # For the account-key, see docs: https://docs.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage?tabs=azure-portal
+    account_key: <azure_blob_access_key>
+    account_name: <azure_account_name>
+    container_name: <azure_storage_bucket_name>
+    use_managed_identity: false
+    # Providing a user assigned ID will override use_managed_identity
+    #user_assigned_id: <user-assigned-identity-id>
+    request_timeout: 0
+    # Configure this if you are using private azure cloud like azure stack hub and will use this endpoint suffix to compose container and blob storage URL. Ex: https://account_name.endpoint_suffix/container_name/blob_name
```
Comment (JStickler) [inline docs/sources/configure/examples/configuration-examples.md:421]:
```suggestion
    # Configure `endpoint_suffix` if you are using private azure cloud like azure stack hub and will use this endpoint suffix to compose container and blob storage URL. Ex: https://account_name.endpoint_suffix/container_name/blob_name
```

File: docs/sources/configure/examples/configuration-examples.md [hunk 2]
```diff
@@ -402,3 +402,47 @@ memberlist:
...
+    #endpoint_suffix: <endpoint-suffix>
+    # If `connection_string` is set, the values of `account_name` and `endpoint_suffix` values will not be used. Use this method over `account_key` if you need to authenticate via a SAS token. Or if you use the Azurite emulator.
```
Comment (JStickler) [inline docs/sources/configure/examples/configuration-examples.md:423]:
```suggestion
    # If `connection_string` is set, the `account_name` and `endpoint_suffix` values will not be used. Use this method over `account_key` if you need to authenticate via an SAS token. Or if you use the Azurite emulator.
```

**Review by JStickler — COMMENTED** [review]
[docs team]  couple of suggestions to clean up the comments.

File: docs/sources/configure/examples/configuration-examples.md [hunk 3]
```diff
@@ -402,3 +402,47 @@ memberlist:
...
+
+# This partial configuration uses Azure for chunk storage
+storage_config:
+  azure:
+    # For the account-key, see docs: https://docs.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage?tabs=azure-portal
+    account_key: <azure_blob_access_key>
+    account_name: <azure_account_name>
+    container_name: <azure_storage_bucket_name>
+    use_managed_identity: false
+    # Providing a user assigned ID will override use_managed_identity
```
[the remark this answers is not in the document]
Comment (paul1r) [reply docs/sources/configure/examples/configuration-examples.md:418]: no, it isn't

**Review by JStickler — APPROVED** [review]
[docs team] LGTM.

**Pull request merged**